File: Python_Collections_benchmark/DictionaryTest.py
from IBenchmark import IBenchmark
from collections import defaultdict


class DictionaryTest(IBenchmark):
    dictionary = defaultdict()
    num_of_operations = 10000

    def __init__(self, num_of_operations):
        self.num_of_operations = num_of_operations

    def add_test(self):
        for i in range(0, self.num_of_operations):
            to_add = len(self.dictionary)
            self.dictionary.setdefault(to_add, to_add)

    def find_test(self):
        to_find = len(self.dictionary) - 1
        for i in range(0, self.num_of_operations):
            result = to_find in self.dictionary.values()

    def remove_test(self):
        for i in range(0, self.num_of_operations):
            to_remove = len(self.dictionary) - 1
            self.dictionary.pop(to_remove)

File: CS_Collections_benchmark/DictionaryTest.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

namespace CS_Collections_benchmark
{
    class DictionaryTest : IBenchmark
    {
        Dictionary<int, int> dictionary;
        int numOfOperations;
        public DictionaryTest()
        {
            dictionary = new Dictionary<int, int>();
            this.numOfOperations = 100;
        }
        public DictionaryTest(int numOfOperations)
        {
            dictionary = new Dictionary<int, int>();
            this.numOfOperations = numOfOperations;
        }

        public void AddTest()
        {
            for (int i = 0; i < numOfOperations; i++)
            {
                int toAdd = dictionary.Count;
                dictionary.Add(toAdd, toAdd);
            }

        }

        public void FindTest()
        {
            int value = dictionary.Count - 1;
            for (int i = 0; i < numOfOperations; i++)
            {
                bool result = dictionary.ContainsValue(value);
            }
        }

        public void RemoveTest()
        {
            int toRemove = dictionary.Count - 1;
            for (int i = 0; i < numOfOperations; i++)
            {
                dictionary.Remove(toRemove);
                toRemove--;
            }
        }
    }
}
